src/components/common/ScrollOverlay/ScrollOverlay.jsx

```diff
@@ -66,42 +60,26 @@ class ScrollOverlay extends Component {
     if (!atTop) {
       classes += styles.notAtTop;
     }
-
     return classes;
   }
 
-  /**
-   * decide if the scroll indicators should be shown or not
-   * @param target - the `.scroll-content` node
-   */
-  updateScrollIndicators(target) {
-    if (!document.body.contains(target)) {
-      return;
-    }
-    // compare using `< 1` instead of `=== 0` because floating point precision
-    const bottomHeight = target.scrollTop + target.clientHeight;
-    const atBottom = Math.abs(bottomHeight - target.scrollHeight) < 1;
-    const atTop = target.scrollTop === 0;
-    this.setState({ atTop, atBottom });
-  }
-
-  render() {
-    const { className, children } = this.props;
-    const { id } = this.state;
-    return (
-      <div
-        className={`scroll-content ${className} ${styles.ScrollOverlay} ${this.stateClasses}`}
-        id={id}
-        ref={this.setDomNode}
-      >
-        { children }
-      </div>
-    );
-  }
+  return (
+    <div
+      className={`scroll-content ${className} ${styles.ScrollOverlay} ${stateClasses()}`}
+      id={id.current}
+      ref={setDomNode}
+    >
+      { children }
+    </div>
+  );
 }
 
+// Static functions & properties
 ScrollOverlay.elementCount = 0;
 
+// eslint-disable-next-line no-plusplus
```

Comment:
Is this maybe a rule that we should exclude?  Otherwise, if we want to keep it, I suggest to rewrite the next line so that it complies. The ++ does not add that much here compared to doing it in an extra line